app/src/main/java/org/openobservatory/ooniprobe/fragment/DashboardFragment.java

```diff
@@ -87,9 +88,12 @@ private void setLastTest() {
 	}
 
 	public void runAll(){
-		Intent intent = RunningActivity.newIntent((AbstractActivity) getActivity(), testSuites);
-		if (intent != null)
-			ActivityCompat.startActivity(getActivity(), intent, null);
+		RunningActivity.runAsForegroundService((AbstractActivity) getActivity(), testSuites);
+		try {
+			((AbstractActivity)getActivity()).bindTestService();
+		} catch (Exception e) {
+			e.printStackTrace();
```

Comment:
Should we do something about exceptions in this case? Perhaps instead of letting them fall-through we should log them to sentry.